protocol/tx_test.go

```diff
@@ -1,4 +1,4 @@
-package protocol
```

Comment:
As I said, that'd be OK in this PR, but we'll only have to undo it in my next PR, when the memstore code starts depending on `protocol` (because the `Snapshot` type is moving there). We'd then have a `prottest/memstore` -> `protocol` -> `prottest` -> `prottest/memstore` dependency cycle.

Would you still like me to create a new `prottest/memstore` even if it doesn't let us get rid of `protocol_test`?